stackslib/src/chainstate/stacks/boot/contract_tests.rs

```diff
@@ -559,6 +561,31 @@ impl HeadersDB for TestSimHeadersDB {
     }
 }
 
+#[test]
+fn pox_4_instantiates() {
+    let mut sim = ClarityTestSim::new();
+    sim.epoch_bounds = vec![0, 1, 2];
+    let delegator = StacksPrivateKey::new();
+
+    let expected_unlock_height = POX_TESTNET_CYCLE_LENGTH * 4;
+
+    // execute past 2.1 epoch initialization
```

Comment:
What is this test supposed to check?  If all it's checking is that pox-4 is syntax-free, then these steps aren't needed -- you can go straight to instantiating pox-4.  If it's instead checking that pox-4 instantiates at the right time, then this test needs to verify that instead.

Reply:
Yes; however, this is not being tested here

Reply:
Then would a pox lockup transition from pox-3 to pox-4 be enough? (As a test.)

Reply:
Yes; you'd need to verify that the `instantiate_epoch_2_5()` and `initialize_epoch_3_0()` functions in `stackslib/src/clarity_vm/clarity.rs` are getting called.

Reply:
OK, and then this test shouldn’t be in this file but instead it should be in `pox_4_tests.rs` yes?

Reply:
Either way is fine with me. If @jbencin already tests this in his PR, then it's fine to omit it here.  However, it's really important that you test the new stacking API.